drivers: watchdog: mcux_wwdt: add driver for NXP LPC 55xxx WWDT

Nephirus opened the pull request:
Adds shim driver for LPC 55xxx Windowed Watchdog Timer (WWDT).

Also updates the "Supported features" table in LPCXpresso55S69
board documentation page.

Signed-off-by: Jiří Keresteš <jiri@kerestes.cz>

Nephirus committed:
drivers: watchdog: mcux_wwdt: add driver for NXP LPC 55xxx WWDT

Adds shim driver for LPC 55xxx Windowed Watchdog Timer (WWDT).

Also updates the "Supported features" table in LPCXpresso55S69
board documentation page.

Fixes #22704.

Signed-off-by: Jiří Keresteš <jiri@kerestes.cz>

Nephirus commented:
Pipeline pass depends on this PR: https://github.com/zephyrproject-rtos/hal_nxp/pull/48

MaureenHelm reviewed (changes requested):
> Pipeline pass depends on this PR: [zephyrproject-rtos/hal_nxp#48](https://github.com/zephyrproject-rtos/hal_nxp/pull/48)

Please add a commit to this PR that updates west.yml to points to the hal PR. See [here](https://docs.zephyrproject.org/latest/guides/modules.html#changes-to-existing-module) for more details. This will help fix the buildkite failure.

Thanks for the contribution!

agansari commented on boards/arm/lpcxpresso55s69/lpcxpresso55s69_ns.dts:
```diff
@@ -79,3 +79,7 @@
 	status = "okay";
 };
+
+&wwdt0 {
```
does it work on lpcxpresso55s69_cpu0 ? If yes, please add it to that.

Nephirus replied:
Thanks for suggestion, I've added relevant bits to lpcxpresso55s69_cpu0.dts and lpcxpresso55s69_cpu0.yaml files.

agansari commented on drivers/watchdog/wdt_mcux_wwdt.c:
```diff
@@ -0,0 +1,186 @@
+}
+
+#define MSEC_TO_WWDT_TICKS(clock_freq, msec) \
+	((uint32_t)(clock_freq * msec / 1000U / 4))
```
Please use the macro MSEC_PER_SEC ore something similar.
Also explain 4 with a comment, i.e. device divides source clock by 4.

Nephirus replied:
Done.

agansari commented on dts/arm/nxp/nxp_lpc55S6x_common.dtsi:
```diff
@@ -210,6 +211,15 @@
+		reg = <0xc000 0x1000>;
+		interrupts = <0 0>;
+		status = "disabled";
+		clk-divider = <1>;
```
Is not the clock divider 4 ? If yes, it should be read int the macro above using DT_PROP()

Nephirus replied:
This clock divider is different from the fixed prescaler in the macro above.

henrikbrixandersen commented:
Please add "Fixes #22704" to the commit introducing the driver.

Nephirus commented:
> Please add "Fixes #22704" to the commit introducing the driver.

Done, thanks!

henrikbrixandersen commented on west.yml:
```diff
@@ -98,7 +98,7 @@ manifest:
       path: tools/net-tools
     - name: hal_nxp
-      revision: 478f03a71256aeb2890e685ebf5a1dc6832a6b5f
+      revision: pull/48/head
```
This needs to be updated to use a SHA before merging.

MaureenHelm replied:
The hal PR was merged, so you can update the SHA now.

Nephirus replied:
Done.

galak commented on soc/arm/nxp_lpc/lpc55xxx/soc.c:
```diff
@@ -77,6 +77,11 @@ static ALWAYS_INLINE void clock_init(void)
 	RESET_PeripheralReset(kHSLSPI_RST_SHIFT_RSTn);
 #endif
 
+#if DT_NODE_HAS_COMPAT_STATUS(DT_NODELABEL(wwdt0), nxp_lpc_wwdt, okay)
```
why us DT macro here v CONFIG_WDT_MCUX_WWDT?

Nephirus replied:
I2C and SPI clocks are both set up using DT macros, so I did the same thing here. I'm not sure if DT macros are better than CONFIG_WDT_MCUX_WWDT option, should I change it?

Nephirus committed:
west.yml: hal_nxp: Use revision with WWDT driver built

The new LPC WWDT shim driver needs HAL update, which is provided by
the commit 7276046215cfcd4ed3d6723f16f7aeb868b0fb60.

Signed-off-by: Jiří Keresteš <jiri@kerestes.cz>

MaureenHelm reviewed (changes requested):
Please remove the intermediate commit `west.yml: hal_nxp: Use PR #48 head to fix pipeline`.

Nephirus commented:
> Please remove the intermediate commit `west.yml: hal_nxp: Use PR #48 head to fix pipeline`.

Done.
